### DISCUSSION TITLE
Studio Development | Local Environment

### DISCUSSION THREAD

**Discussion by nilooy** [supabase/supabase#8047] [Questions]
I'm trying run `studio` on development environment

I have followed these guides 
https://github.com/supabase/supabase/blob/master/DEVELOPERS.md 
https://github.com/supabase/supabase/blob/master/studio/README.md
https://supabase.com/docs/guides/hosting/docker
to setup,
Steps Taken:

1. forked and cloned
2. Ran
```sh
cd supabase/studio
npm Install
```
3. Ran
```sh 
cd ../docker
cp .env.example .env
docker-compose -f docker-compose.yml -f ./dev/docker-compose.dev.yml up
```
5. As .env and its data already there on the studio folder, I ran  
```sh
cd ../studio
npm run dev 
```
It is stuck on the loading screen with some console errors on this page http://0.0.0.0:8082/project/default
> but http://0.0.0.0:3000/project/default run fine but there are no local changes reflecting on this container.

Reposting from https://discord.com/channels/839993398554656828/1004321585508917278/1004322324251680819

**Comment by Hallidayo** [comment]
Hi @nilooy,

It seems you closed an issue #8023 which is related, Can I check is this working for you now?

**Reply by nilooy** [reply]
They are not related. The issue was to install the deps but it was my mistake but after installing the deps, so I closed that but when I tried running studio in dev mode along with docker containers, the UI was just stuck in the loading screen.

But I tried today  with a fresh setup, and it worked. 💯

**Reply by jolo-dev** [reply]
Hmm...I ran into the same issue and couldn't resolve it.

**Reply by nilooy** [reply]
@jolo-dev Can you add more information about the issue? i can try to help.

**Reply by jolo-dev** [reply]
Hey @nilooy,
Well, I simply followed the Quickstart from the `studio/README.md`.

```sh
npm i # install dependencies
npm run dev # start dev server
npm run test # run tests
npm run -- --watch # run tests in watch mode
```

I also tried your steps above.
But the front page is stuck and keeps loading.

Which node are you using? It seems like the studio needs `v14` but in `DEVELOPER.md` is written that you need `v16`.
Either way, it doesn't work. However, I use [Gitpod](https://gitpod.io) for spinning up my dev environment and wanna try it locally.

**Reply by jolo-dev** [reply]
And locally I get this. I think this is a MacOS M1 problem.

![Screenshot 2022-10-06 at 12 29 20](https://user-images.githubusercontent.com/54506108/194290948-58d6630a-d044-45fb-a751-5714105c67d5.png)
